ERCS/erc-7628.md

```diff
@@ -0,0 +1,111 @@
+---
+eip: 7628
+title: ERC-721 Balance Extension
+description: Extend ERC-721 to include balance functionalities similar to ERC-20.
+author: Chen Liaoyuan (@chenly) <cly@kip.pro>
+discussions-to: https://ethereum-magicians.org/t/erc-7628-erc-721-balance-extension/18744
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-02-20
+requires: 20, 721
+---
+
+## Abstract
+
+This extension defines an interface that adds balance functionalities to [ERC-721](./eip-721.md) tokens, enabling functions and events for balance querying, transfer, and approval similar to [ERC-20](./eip-20.md) tokens.
```

Comment:
I'm assuming the functionality you're talking about is separate from the functions that already exist on ERC-721, so please add some text explaining exactly what new functionality you want to add is.

Is this similar to [ERC-1155](https://eips.ethereum.org/EIPS/eip-1155) where you can mix fungible and non-fungible tokens in the same contract?

Reply:
The proposal aims to find a middle ground between ERC-721 and ERC-1155. While ERC-1155 can represent the share of an asset held by a particular address, tokens lack uniqueness. On the other hand, ERC-721 provides uniqueness for each token, but they lack attributes to represent the share of an asset.